metricbeat/module/kafka/partition/_meta/data.json

```diff
@@ -26,11 +27,13 @@
                 "replica": 0
             },
             "topic": {
-                "name": "foo-1532945633-185372965"
-            }
+                "name": "foo-1538389014-739473801"
+            },
+            "topic_broker_id": "0-foo-1538389014-739473801-0",
```

Comment:
Why not `topic.id` and `topic.broker_id`?

Reply:
I am not really sure about the naming and placing of these composed ids, in demo they are directly at the module level as `kafka.topic_partition_broker_id`, I moved them to partition as in some way they identify the partition. For the same reason I'd keep them as fields directly under the partition, and I wouldn't expect other fields under `kafka.partition.topic`.  `kafka.partition.topic.id` may be seen also as if it should be the same as `kafka.topic.id`.

In any case I don't have a strong opinion on these names, so as you prefer.

Reply:
I see now that `topic.id` shows up in different events. So `kafka.topic.id` would probably make most sense? Meaning that we should move out `topic` from partition to also have `kafka.topic.name` (would be a breaking change)?

Reply:
`kafka.topic.name` was added in #7767 to replace `kafka.partition.topic.name` and `kafka.consumergroup.topic.name` as a common name. These fields contain the topic name as seen by kafka, old ones are marked as deprecated.

The fields added in this PR are created by us to help on queries and don't exist in kafka, they identify the partition in the topic and in the broker (there can be two partitions with the same id, but of different topics, and two partitions with the same id and different topics and/or brokers), this is why I'd add them under `kafka.partition`, I see them more like identifiers of the partition.

Reply:
Is it possible that that more then one topic.id shows up in a single event?

Reply:
Not sure of following you... a single event will have an only `kafka.topic.id` and an only `kafka.partition.topic_id` (or however we call it), and both ids will be different, being the second one composed by the first one and `kafka.partition.id`.

Reply:
That actually answers my questions. There are 2 topic.id in 1 event so there have to be different fields like you suggested.